testgrid/cmd/entomologist/main.go

```diff
@@ -0,0 +1,212 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/test-infra/pkg/io"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/testgrid/issue_state"
+)
+
+type githubClient interface {
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+	ListIssues(org, repo string) ([]github.Issue, error)
+	ListIssueComments(org, repo string, number int) ([]github.IssueComment, error)
+}
+
+const defaultPollTime = 1 * time.Hour
+
+type options struct {
+	ghProxyEndpoint string
+	ghCredentials   []byte
+	organization    string
+	repository      string
+	gcsPath         string
+	gcsCredentials  string
+	oneshot         bool
+}
+
+func (o *options) parseArgs(fs *flag.FlagSet, args []string) error {
+
+	var ghCredInput string
+	fs.StringVar(&o.ghProxyEndpoint, "github-proxy", "https://api.github.com", "Address of GitHub proxy (optional)")
+	fs.StringVar(&ghCredInput, "github-token", "", "GitHub OAuth credentials (optional)")
+	fs.StringVar(&o.organization, "github-org", "", "GitHub organization")
+	fs.StringVar(&o.repository, "github-repo", "", "GitHub repository")
+	fs.StringVar(&o.gcsPath, "output", "", "GCS output (gs://bucket)")
+	fs.StringVar(&o.gcsCredentials, "gcs-cred", "", "/path/to/service/account/credentials (as .json)")
+	fs.BoolVar(&o.oneshot, "oneshot", false, "Write proto once and exit instead of monitoring GitHub for changes")
+	fs.Parse(args)
+
+	o.ghCredentials = []byte(ghCredInput)
+
+	return o.validate()
+}
+
+func (o *options) validate() error {
+	if o.organization == "" || o.repository == "" {
+		return errors.New(fmt.Sprintf("%s/%s is not a valid GitHub location", o.organization, o.repository))
+	}
+	if o.gcsPath == "" || !strings.HasPrefix(o.gcsPath, "gs://") {
+		return errors.New("invalid or missing --output")
+	}
+	if o.gcsCredentials == "" {
+		return errors.New("--gcs-cred required for write operations")
+	}
+	return nil
+}
+
+func parseOptions() options {
+	var o options
+
+	if err := o.parseArgs(flag.CommandLine, os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("Invalid Flags")
+	}
+
+	return o
+}
+
+func main() {
+	opt := parseOptions()
+
+	getToken := func() []byte {
+		return opt.ghCredentials
+	}
+
+	ghct := github.NewClient(getToken, "", opt.ghProxyEndpoint)
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	client, _ := io.NewOpener(ctx, opt.gcsCredentials)
+	if client == nil {
+		logrus.Fatalf("Empty Credentials (at %s) not allowed for write operation", opt.gcsCredentials)
+	}
+
+	gcsFile := fmt.Sprintf("/bugs-%s-%s", opt.organization, opt.repository)
```

Comment:
Is this filename schema arbitrary, or is this something expected by TestGrid?

Reply:
It's expected by TestGrid, but might change in the near future

Reply:
OK, in that case can we leave a comment explaining as much? eg: 
```
// testgrid expects issue_state.proto files for org/repo to be be named bugs-org-repo
```

Reply:
Done